[clang] Make source locations space usage diagnostics numbers easier to read

bricknerb opened the pull request:
Instead of writing "12345678B", write "12345678B (12.34MB)".

bricknerb committed:
[clang] Make source locations space usage diagnostics numbers easier to read
Instead of write "12345678B", write "12345678B (12.34MB)".

bricknerb committed:
[clang] Fix variable names to PascalCase

bricknerb commented on clang/lib/Basic/SourceManager.cpp:
```diff
@@ -2227,6 +2227,28 @@ LLVM_DUMP_METHOD void SourceManager::dump() const {
+       {1'000'000UL, 'M'},
+       {1'000UL, 'k'}}};
+
+  std::string human_string;
```
Oops!
Done.

ilya-biryukov commented:
Do we really care about the exact byte numbers? Maybe we should only show the human-friendly version?
It's appealing to have less noise if we can.

bricknerb commented:
> Do we really care about the exact byte numbers? Maybe we should only show the human-friendly version? It's appealing to have less noise if we can.

Yes, I was considering both options.
It might be useful to see the full number in case you want to diff between logs, and the diff would be relatively small (people might care about zero diff vs. tiny diff), so I decided to make this change not lose any information.

ilya-biryukov commented on clang/lib/Basic/SourceManager.cpp:
```diff
@@ -2227,6 +2227,28 @@ LLVM_DUMP_METHOD void SourceManager::dump() const {
   }
 }
 
+static std::string NumberToHumanString(uint64_t Number) {
```
LLVM Style guide asks to use `camelCase` and a verb phrase (see https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly). `Number` also seems a bit redundant, because it's the first parameter anyway.

Let's turn this to `camelCase`?
Extra suggestion (optional): Maybe rename to `toHumanString`?

bricknerb replied:
Renamed to `humanizeNumber()`.

ilya-biryukov commented:
> Yes, I was considering both options. It might be useful to see the full number in case you want to diff between logs, and the diff would be relatively small (people might care about zero diff vs. tiny diff), so I decided to make this change not lose any information.

Makes sense, and given that it's a relatively infrequently surfaced feature, I think it's fine to change our mind on that later.

ilya-biryukov commented on clang/lib/Basic/SourceManager.cpp:
```diff
@@ -2227,6 +2227,28 @@ LLVM_DUMP_METHOD void SourceManager::dump() const {
+  llvm::raw_string_ostream HumanStringStream(HumanString);
+  for (const auto &[UnitSize, UnitSign] : Units) {
+    if (Number >= UnitSize) {
+      HumanStringStream << llvm::format(
```
Suggestion: use `formatv` and it can convert to string directly.
```cpp
HumanString += llvm::formatv(...);
```

bricknerb replied:
Thanks, done.

ilya-biryukov commented on clang/lib/Basic/SourceManager.cpp:
```diff
@@ -2227,6 +2227,28 @@ LLVM_DUMP_METHOD void SourceManager::dump() const {
+    if (Number >= UnitSize) {
+      HumanStringStream << llvm::format(
+          "%.2f%c", Number / static_cast<double>(UnitSize), UnitSign);
+      break;
```
Suggestion: maybe reduce the amount of the code a little by using `return` instead of `break`?
That would also make the extra variable unnecessary.

```cpp
for (... : Units) {
  if (Number >= UnitSize)
    return llvm::formatv(...);
}
return std::to_string(Number);
```

bricknerb replied:
Done.

bricknerb committed:
[clang] Rename NumberToHumanString() to humanizeNumber() and add a comment

bricknerb committed:
[clang] Simplify humanizeNumber()
1. llvm::formatv() instead of llvm::format().
2. Early return.
3. Use std::to_string() when no special formatting is necessary.

ilya-biryukov reviewed (approved):
LGTM, thanks!
